lvmd/vgservice.go

```diff
@@ -148,33 +148,19 @@ func (s *vgService) send(server proto.VGService_WatchServer) error {
 	}
 	res := &proto.WatchResponse{}
 	for _, vg := range vgs {
-		vgFree, err := vg.Free()
-		if err != nil {
-			return status.Error(codes.Internal, err.Error())
-		}
-		vgSize, err := vg.Size()
+		pools, err := vg.ListPools()
 		if err != nil {
 			return status.Error(codes.Internal, err.Error())
 		}
-		dc, err := s.dcManager.FindDeviceClassByVGName(vg.Name())
-		if err == ErrNotFound {
-			continue
-		}
-		if err != nil {
-			return err
-		}
 
-		switch dc.Type {
-		case TypeThick:
-			if dc.Default {
-				res.FreeBytes = vgFree
+		for _, pool := range pools {
+			dc, err := s.dcManager.FindDeviceClassByPoolName(vg.Name() + "/" + pool.Name())
```

Comment:
If arguments are required to be concatenated, it is better to express the restriction as a function parameter.
So, why not define the function as  `FindDeviceClassByPoolName(vg string, pool string)` ?

Reply:
Ah! Makes sense.